Implemented Tap/Double Tap/Hold/Don't Tap

carterjmoore opened the pull request:
- Implemented tap, double tap, hold, and don't tap in the main game loop
- Added a test for tap, double tap, and hold
- Moved the code for displayInstruction() to the InstructionUtil class, as it was getting pretty big
- Added a final variable to control default text size for all displayed instructions
- Upped the default text size for effect with a small "DON'T" text (you'll see what I mean when you get the Don't Tap instruction. We can make this a little smaller though.

Also, I've had it happen twice where the Don't Tap instruction will display its text twice, but haven't had it happen in a while, and can't find out what causes it, so just keep that in mind.

carterjmoore committed:
Implemented tap/double tap/hold

carterjmoore committed:
Implemented Don't Tap

carterjmoore committed:
Moved displayInstruction() to util

The displayInstruction() method was getting pretty big with all the new instructions being added, so I moved it to InstructionUtil.java

carterjmoore committed:
Changed some text sizes

Added a local final variable for text size, and changed it to be bigger (can change this later), and made the text for "Don't" in the don't tap instruction smaller.

fredy20501 commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/utils/Instruction.java:
```diff
@@ -3,6 +3,9 @@
 public enum Instruction {
     BUTTON, //Temp instruction for testing
     TAP,
+    DOUBLETAP,
+    HOLD,
+    DONTTAP,
```
I suggest we rename the instructions as `DOUBLE_TAP` and `DONT_TAP` to make them more readable.

carterjmoore replied:
Good idea, I'll change that

fredy20501 commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/activities/tests/TapTestActivity.java:
```diff
@@ -0,0 +1,69 @@
+        newInstruction();
+    }
+
+    private void newInstruction(){
+        currentInstruction = instructions[r.nextInt(instructions.length)];
+        switch(currentInstruction){
+            case TAP:
+                tapText.setText("Tap: " + numInputs);
+                break;
+            case DOUBLETAP:
+                tapText.setText("Double Tap: " + numInputs);
+                break;
+            case HOLD:
+                tapText.setText("Hold: " + numInputs);
+                break;
+            default:
+                tapText.setText("");
+                break;
+        }
+    }
```
I find that the way we update the label here is very confusing, and makes it unclear what the number (numInputs) actually means. I made PR #9 that includes changes to make it clearer. That PR is set to merge into this branch, so if you agree you can just merge it right away (it will get reviewed by Hailey once it is in this branch).

carterjmoore replied:
Alright makes sense, I went ahead and merged it

fredy20501 commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/activities/GameActivity.java:
```diff
@@ -138,66 +145,44 @@ public void onTick(long l) {
 
     //Handle any inputs received
     private void detectInput(Instruction instruction) {
+        //If instruction is don't tap and a tap input is received, end game
+        if(currentInstruction == Instruction.DONTTAP  &&
+                (instruction == Instruction.TAP ||
+                 instruction == Instruction.DOUBLETAP ||
+                 instruction == Instruction.HOLD)) {
+            endGame();
+            return;
+        }
```
Here you can just check for `Instruction.TAP` since a `TAP` event is always triggered as part of `DOUBLETAP` and `HOLD`. Like this:
```java
if(currentInstruction == Instruction.DONTTAP && instruction == Instruction.TAP) {...}
```

carterjmoore replied:
Good point, I didn't consider this. I'll do that.

fredy20501 reviewed (approved):
Looks great!

fredy20501 committed:
Update TapTestActivity (#9)

carterjmoore committed:
Implemented Fred's Suggestions

Also moved the label.setSize() and setAlignment() to outside the cases, so that they only happen once.

carterjmoore committed:
Resolved Merge Conflict

carterjmoore committed:
Fixed bug that accepted multiple inputs at once

fredy20501 commented on app/src/main/java/ca/unb/mobiledev/reflexrevolution/utils/InstructionUtil.java:
```diff
@@ -19,16 +25,81 @@
+                layout.addView(label);
+                break;
+
+            case DOUBLE_TAP:
+                label.setText("DOUBLE\nTAP");
+                layout.addView(label);
+                break;
+
+            case HOLD:
+                label.setText("HOLD");
+                layout.addView(label);
+                break;
```
For the double-tap and hold instructions, it might be cool to use the same format as the don't tap. What I mean is that all of these instructions would have the main label "TAP" with a secondary label saying either "don't", "double", "hold" in a smaller font above.

That way the player really needs to pay attention since they will all look similar.

carterjmoore replied:
Oh I like that idea! I'll throw it in.

carterjmoore committed:
Made "HOLD" and "DOUBLE" small text

hsavoie reviewed (approved):
Looks good to me too!
